Marlin/ultralcd.cpp

```diff
@@ -110,6 +110,9 @@ uint8_t lcdDrawUpdate = LCDVIEW_CLEAR_CALL_REDRAW; // Set when the LCD needs to
   #if HAS_POWER_SWITCH
     extern bool powersupply;
   #endif
+  #if HAS_CASE_LIGHT
+    extern bool caselight;
+  #endif
```

Comment:
Instead of two flags for the same thing, just use `extern bool case_light_on` and make `case_light_on` a global.